mlflow/sparkml.py

```diff
@@ -0,0 +1,116 @@
+"""
+MLflow integration for SparkML models.
+
+SparkML models are saved and loaded using native SparkML persistence.
+The models can be exported as pyfunc for out-of Spark deployment or it can be loaded back as Spark
+Transformer in order to score it in Spark. The pyfunc flavor instantiates SparkContext internally
+and reads the input data as Spark DataFrame prior to scoring.
+
+Note: SparkML model can not be exported as spark_udf as those two formats are incompatible.
+"""
+
+from __future__ import absolute_import
+
+import os
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.environment import CondaEnvironment
+from mlflow.models import Model
+
+import pyspark
+from pyspark.ml.pipeline import PipelineModel
+from pyspark.ml.base import Transformer
+
+FLAVOR_NAME = "sparkml"
+
+
+def log_model(spark_model, artifact_path, env, jars=None):
+    """
+    Log model using supplied flavor.
+
+    :param spark_model: Model to be saved.
+    :param artifact_path: Run-relative artifact path.
+    :param env: Conda environment, defaults to minimum viable environment.
+    :param jars: List of jars needed by the model.
+    """
+    return Model.log(artifact_path=artifact_path, flavor=mlflow.sparkml, spark_model=spark_model,
+                     jars=jars, env=env)
+
+
+def save_model(spark_model, path, mlflow_model=Model(), env=CondaEnvironment(), jars=None):
+    """
+    Save SparkML model at given local path.
+
+    Uses SparkML persistence mechanism.
+
+    :param spark_model: Model to be saved.
+    :param path: Local path where the model is to be save.
+    :param mlflow_model: MLflow model config.
```

Comment:
`mlflow_model` is to add this to an existing model, right? You should document that and maybe also make the parameter name consistent in `sklearn.save_model` since it's called `model` there. Also I'd move this to be the last argument since it seems unlikely that users would directly pass this (since they'll use log_model instead).

Reply:
Yes, I add documentation.  I actually renamed the sklearn one too. I though mlflow_model was more clear than just model.